Documentation/discovery_protocol.md

```diff
@@ -0,0 +1,107 @@
+This document talks about the technical details of discovery service protocol. It helps users to learn about how discovery service works and what behavior to expect.
+
+# Discovery Service Protocol
+
+Discovery service protocol helps new etcd member to discover all other members in cluster bootstrap phase using a shared discovery URL.
+
+Discovery service protocol is _only_ used in cluster bootstrap phase, and cannot be used for runtime reconfiguration or cluster monitoring.
+
```

Comment:
we also need to mention you need a new token if the previous token is used (even if the bootstrap failed.)